src/main/java/me/lucko/fabric/api/permissions/v0/Options.java

```diff
@@ -0,0 +1,213 @@
+ *  Copyright (c) contributors
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ */
+
+package me.lucko.fabric.api.permissions.v0;
+
+import net.minecraft.command.CommandSource;
+import net.minecraft.entity.Entity;
+
+import org.jetbrains.annotations.Contract;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * A simple options (metadata) API.
+ */
+public interface Options {
+
+    /**
+     * Gets the value of an option for the given source.
+     *
+     * @param source the source
+     * @param key the option key
+     * @return the option value
+     */
+    static @NotNull Optional<String> get(@NotNull CommandSource source, @NotNull String key) {
+        Objects.requireNonNull(source, "source");
+        Objects.requireNonNull(key, "key");
+        return OptionRequestEvent.EVENT.invoker().onOptionRequest(source, key);
+    }
+
+    /**
+     * Gets the value of an option for the given source, falling back to the {@code defaultValue}
+     * if nothing is returned from the provider.
+     * 
+     * @param source the source
+     * @param key the option key
+     * @param defaultValue the default value to use if nothing is returned
+     * @return the option value
+     */
+    @Contract("_, _, !null -> !null")
+    static String get(@NotNull CommandSource source, @NotNull String key, String defaultValue) {
+        return get(source, key).orElse(defaultValue);
+    }
+
+    /**
+     * Gets the value of an option for the given source, and runs it through the given {@code valueTransformer}.
+     *
+     * <p>If nothing is returned from the provider, an {@link Optional#empty() empty optional} is returned.
+     * (the transformer will never be passed a null argument)</p>
+     *
+     * <p>The transformer is allowed to throw {@link IllegalArgumentException} or return null. This
```

Comment:
I would've loved to see checked exceptions used here because I wouldn't think retrieving options would throw. 0.2 is already out, but just my 2 cents.